paddle/fluid/operators/mkldnn/pool_mkldnn_op.cc

```diff
@@ -28,6 +28,270 @@ using mkldnn::reorder;
+                                unique_name)) {
+    if (!this->isBwdCached()) {
+      PADDLE_ENFORCE_EQ(in_x->layout(), DataLayout::kMKLDNN,
+                        platform::errors::InvalidArgument(
+                            "Wrong layout set for Input tensor"));
+      PADDLE_ENFORCE_NE(in_x->format(), MKLDNNMemoryFormat::undef,
+                        platform::errors::InvalidArgument(
+                            "Wrong format set for Input tensor"));
+
+      PADDLE_ENFORCE_EQ(out_grad->layout(), DataLayout::kMKLDNN,
+                        platform::errors::InvalidArgument(
+                            "Wrong layout set for Input output_grad tensor"));
+      PADDLE_ENFORCE_NE(out_grad->format(), MKLDNNMemoryFormat::undef,
+                        platform::errors::InvalidArgument(
+                            "Wrong format set for Input output_grad tensor"));
+
+      PADDLE_ENFORCE_EQ(
+          ctx.Attr<bool>("is_test"), false,
+          platform::errors::InvalidArgument(
+              "is_test attribute should be set to False in training phase."));
+
+      std::string pooling_type = ctx.Attr<std::string>("pooling_type");
+
+      std::vector<int> ksize_temp = ctx.Attr<std::vector<int>>("ksize");
+      std::vector<int64_t> ksize(begin(ksize_temp), end(ksize_temp));
+
+      std::vector<int> strides_temp = ctx.Attr<std::vector<int>>("strides");
+      std::vector<int64_t> strides(begin(strides_temp), end(strides_temp));
+
+      std::vector<int> paddings_temp = ctx.Attr<std::vector<int>>("paddings");
+      std::vector<int64_t> paddings(begin(paddings_temp), end(paddings_temp));
+
+      bool global_pooling = ctx.Attr<bool>("global_pooling");
+      std::string padding_algorithm =
+          ctx.Attr<std::string>("padding_algorithm");
+
+      auto in_x_dims = in_x->dims();
+      framework::DDim data_dims =
+          framework::slice_ddim(in_x_dims, 2, in_x_dims.size());
+
+      if (global_pooling) {
+        operators::UpdateKsize(&ksize, data_dims);
+      }
+
+      operators::UpdatePadding(&paddings, global_pooling, 0, padding_algorithm,
+                               data_dims, strides, ksize);
+
+      auto src_tz = paddle::framework::vectorize<int64_t>(in_x->dims());
+      auto diff_src_tz =
+          paddle::framework::vectorize<int64_t>(in_x_grad->dims());
+      auto diff_dst_tz =
+          paddle::framework::vectorize<int64_t>(out_grad->dims());
+
+      auto diff_dst_md = mkldnn::memory::desc(
+          diff_dst_tz, platform::MKLDNNGetDataType<T>(), out_grad->format());
+      auto diff_src_md =
+          mkldnn::memory::desc(diff_src_tz, platform::MKLDNNGetDataType<T>(),
+                               MKLDNNMemoryFormat::any);
+
+      auto mkldnn_paddings = platform::ToMkldnnPadding(paddings);
+      const bool ceil_mode = ctx.Attr<bool>("ceil_mode");
+
+      if (ceil_mode) {
+        CorrectOutputSize(src_tz, diff_dst_tz, ksize, paddings, strides,
```

Comment:
`PoolOp::InferShape` has calculated outputsize, this function `CorrectOutputSize` is for calculating mkldnn_paddings[1] ?

Reply:
@lidanqing-intel Yes. I just move this Handler code from mkldnn_reuse.h to pool_mkldnn_op.cc  as mkldnn_reuse.h got too big, so I haven't done added CorrectOutputSize() , but it does modify mkldnn padding.